### PR TITLE
fix: [IAC-106]: Cleanup: Create stack modal issues

### PR DESCRIPTION
**Pull request by stuart-mcdonnell**
### Summary

- Use correct images in create stack modal for provisioner types
- Disable other options other than terraform and add coming soon text below
- Add git fetch type to repo details screen:
- - gitFetchType key that can be either branch or commitId
- On repo details screen the repo is always asked for even when connector is of repo type, only needed when account type connector
- Remove harness store from selection
- Remove tags from the form
- - Name and description only

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by stuart-mcdonnell** [comment]
retrigger jest

**Comment by stuart-mcdonnell** [comment]
retrigger cypress-cd

**Comment by stuart-mcdonnell** [comment]
retrigger cypress-pipeline

**Review by willd0yle — DISMISSED** [review]

**Comment by stuart-mcdonnell** [comment]
retrigger jest

File: src/modules/85-iacm/components/IACMResourceStackWizard/StackProvisionerDetailsStep.tsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ const ProvisionerDetailsStage: React.FC<StackWizardStepProps> = props => {
                     <DateInput
                       name="ttl"
                       timePrecision="minute"
+                      value={String(values.ttl)}
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMResourceStackWizard/StackProvisionerDetailsStep.tsx:80]: We really must get `DateInput` hooked up to Formik in UICore.

File: src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx [hunk 2]
```diff
@@ -12,6 +12,11 @@ import { useStrings } from 'framework/strings'
 import type { StackWizardStepProps } from './index'
 import css from './StackWizard.module.scss'
 
+export enum GitFetchTypes {
+  Branch = 'branch',
+  Commit = 'commitId'
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx:17]: *SUPER minor and totally ignorable*, but a good convention is to use all caps for enums
```suggestion
  BRANCH = 'branch',
  COMMIT = 'commitId'
```

File: src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx [hunk 3]
```diff
@@ -24,27 +29,51 @@ const StackRepoDetailsStep: React.FC<StackWizardStepProps> = (props): JSX.Elemen
...
+            then: Yup.string().required(getString('common.git.validation.repoRequired')),
+            otherwise: Yup.string().notRequired()
+          }),
+          scriptsPath: Yup.string().required(getString('iacm.stackWizard.scriptsPathRequired')),
+          gitFetchType: Yup.string().required(getString('validation.thisIsARequiredField')),
+          branch: Yup.string().when('gitFetchType', {
+            is: value => value === GitFetchTypes.Branch,
+            then: Yup.string().required(getString('common.git.validation.branchRequired')),
+            otherwise: Yup.string().required(getString('validation.commitId'))
+          })
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx:50]: Nice YUPing!

File: src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx [hunk 4]
```diff
@@ -24,27 +29,51 @@ const StackRepoDetailsStep: React.FC<StackWizardStepProps> = (props): JSX.Elemen
...
-                  <FormInput.MultiTextInput name="scriptsPath" label={getString('iacm.stackWizard.scriptsPath')} />
+                  {values?.repoConnectorObject?.scope === 'account' && (
+                    <FormInput.Text name="repo" label={getString('repository')} />
+                  )}
+                  <FormInput.Select
+                    name="gitFetchType"
+                    label={getString('pipeline.manifestType.gitFetchTypeLabel')}
+                    items={[
+                      { label: getString('gitFetchTypes.fromBranch'), value: 'branch' },
+                      { label: getString('gitFetchTypes.fromCommit'), value: 'commitId' }
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoDetailsStep.tsx:69]: Better to use your enum here to ensure consistency
```suggestion
                      { label: getString('gitFetchTypes.fromBranch'), value: GitFetchTypes.Branch },
                      { label: getString('gitFetchTypes.fromCommit'), value: GitFetchTypes.Commit }
```

File: src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoTypeStep.tsx [hunk 5]
```diff
@@ -103,24 +99,29 @@ const StackRepoTypeStep: React.FC<StepProps<StackWizardStepProps>> = props => {
...
                       multiTypeProps={{ expressions }}
-                      setRefValue
+                      onChange={(value: any) => {
+                        /* istanbul ignore next */
+                        const connectorValue =
+                          /* istanbul ignore next */
+                          value?.scope && value?.scope !== 'project'
+                            ? `${value.scope}.${value?.record?.identifier}`
+                            : value?.record?.identifier || value
+                        formik?.setFieldValue('repoConnector', connectorValue)
```
Comment (knagurski) [inline src/modules/85-iacm/components/IACMResourceStackWizard/StackRepoTypeStep.tsx:123]: Might as well destructure `setFieldValue` like you did with `values` and `isValid`

**Review by knagurski — DISMISSED** [review]
Just some minor suggestions. Looks good though

**Comment by stuart-mcdonnell** [comment]
retrigger coverage

**Comment by stuart-mcdonnell** [comment]
retrigger jest

**Comment by stuart-mcdonnell** [comment]
retrigger cypress-pipeline

**Comment by stuart-mcdonnell** [comment]
retrigger coverage

**Review by willd0yle — APPROVED** [review]

**Review by knagurski — APPROVED** [review]
Looks good. One wee point to think about is the look of `Heading`; the `level` should be set according to the hierarchy of the page and the context the heading is in, but you can and should use variations to achieve the correct look, e.g. `font={{ variation: FontVariation.H2 }}`. The `level` prop is to ensure it's correct for screen readers, not for styling.

**Pull request merged**